jpos/src/main/java/org/jpos/security/ModeOfUse.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * jPOS Project [http://jpos.org]
+ * Copyright (C) 2000-2019 jPOS Software SRL
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.jpos.security;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Defines the operation that the key contained in the key block can perform.
+ * <p>
+ * Each value represents byte 8 of the Keyblok Header.
+ */
+public enum ModeOfUse {
+
+    /**
+     * The key may be used to perform both encrypt and decrypt operations.
+     */
+    ENCDEC          ('B', "Encryption and Decryption"),
+
+    /**
+     * The key may be used to perform MAC calculation <i>(both generate &
+     * verify)</i> operations.
+     */
+    GENVER          ('C', "Verification and Genaration of MAC, CVD"),
```

Comment:
Watch out super minor typo (Generation)

Reply:
Heh I have to write "Feel free to submit your comments"